From 88bf3313850bc7f0e6db21daee2b8d8e607b7bb4 Mon Sep 17 00:00:00 2001
From: Samuel Martin <s.martin49@gmail.com>
Date: Sat, 8 Feb 2014 14:33:27 +0100
Subject: [PATCH] configure.ac: add check for libgcrypt-config program

This allows to override the default libgcrypt-config location (which is
useful when cross-compiling), instead of using the one from the host
system.

Signed-off-by: Samuel Martin <s.martin49@gmail.com>
---
 configure.ac | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 02fb8aa..5a2267b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3943,16 +3943,17 @@ dnl
 AC_ARG_ENABLE(libgcrypt,
   [  --disable-libgcrypt     gcrypt support (default enabled)])
 AS_IF([test "${enable_libgcrypt}" != "no"], [
+  AC_PATH_PROG(GCRYPT_CONFIG,libgcrypt-config,libgcrypt-config)
   AC_CHECK_DECL([GCRYCTL_SET_THREAD_CBS], [
-    libgcrypt-config --version >/dev/null || \
+    ${GCRYPT_CONFIG} --version >/dev/null || \
         AC_MSG_ERROR([gcrypt.h present but libgcrypt-config could not be found])
     AC_CHECK_LIB(gcrypt, gcry_control, [
       have_libgcrypt="yes"
-      GCRYPT_CFLAGS="`libgcrypt-config --cflags`"
-      GCRYPT_LIBS="`libgcrypt-config --libs`"
+      GCRYPT_CFLAGS="`${GCRYPT_CONFIG} --cflags`"
+      GCRYPT_LIBS="`${GCRYPT_CONFIG} --libs`"
     ], [
       AC_MSG_ERROR([libgcrypt not found. Install libgcrypt or use --disable-libgcrypt. Have a nice day.])
-    ], [`libgcrypt-config --libs`])
+    ], [`${GCRYPT_CONFIG} --libs`])
   ], [
     AC_MSG_ERROR([libgcrypt version 1.1.94 or higher not found. Install libgcrypt or use --disable-libgcrypt. Have a nice day.])
   ], [#include <gcrypt.h>]
-- 
1.8.5.4

